common/workunit/workunit.cpp

```diff
@@ -6329,17 +6331,47 @@ void CLocalWorkUnit::setDebugAgentListenerIP(const char * ip)
     p->setProp("@DebugListenerIP", ip);
 }
 
-IStringVal& CLocalWorkUnit::getSecurityToken(IStringVal &str) const
+void CLocalWorkUnit::getWorkunitDistributedAccessToken(IStringVal & datoken) const
 {
     CriticalBlock block(crit);
-    str.set(p->queryProp("@token"));
-    return str;
+    datoken.set(p->queryProp("@distributedAccessToken"));
 }
 
-void CLocalWorkUnit::setSecurityToken(const char *value)
+bool CLocalWorkUnit::setDistributedAccessToken(const char * user)
 {
     CriticalBlock block(crit);
-    p->setProp("@token", value);
+
+    //If this format changes, you must update the isWorkunitDAToken() and extractFromWorkunitDAToken() methods
+    VStringBuffer datoken("HPCC[u=%s,w=%s]", user, queryWuid());
+
+    IDigitalSignatureManager * pDSM = queryDigitalSignatureManagerInstanceFromEnv();
+    if (pDSM && pDSM->isDigiSignerConfigured())
+    {
+        const char * pFN = strrchr(pDSM->queryPrivateKeyFile(), '/');//isolate private key file name
+        if (!pFN)
+            pFN = pDSM->queryPrivateKeyFile();
+        else
+            ++pFN;
+        datoken.append(pFN).append(';');//include private key file name used to encrypt
```

Comment:
Is there a JIRA open about this identifying which public key to use to verify, rather than the name of the private key being used?  So that if we want to we can allow the public keys to live longer?

Reply:
@jakesmith @RussWhitehead Yes, the concept looks good.  I wasn't sure it would make it into 7.0 but its great that it is, since its a major bit of configuration for these features.

I would think that the keypairName could be configured for signing workunits, and used as the keyname part of the token.  The names could then be versioned if we wanted to keep them around for verification even after they are no longer used for signing.